apps/web/src/pages/templates/components/CreateWorkflowDropdown.tsx

```diff
@@ -38,8 +45,29 @@ export const CreateWorkflowDropdown = ({
   onTemplateClick: (template: IBlueprintTemplate) => void;
   onAllTemplatesClick: React.MouseEventHandler<HTMLButtonElement>;
 }) => {
+  const isV2Enabled = useFeatureFlag(FeatureFlagsKeysEnum.IS_V2_ENABLED);
   const segment = useSegment();
   const { item: templateId, onMouseEnter, onMouseLeave } = useHoverOverItem<string>();
+  const { toggle, setPath, Component } = useDocsModal();
+
+  useEffect(() => {
+    if (!isV2Enabled) {
+      return;
+    }
+    setPath(PATHS[ROUTES.STUDIO_FLOWS] as string);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [isV2Enabled]);
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
❓ **question:** Is there a reason we omitted `setPath`? We should avoid omitting dependencies unless there's a specific concern‏